firefox_ui_tests/remote/security/test_safe_browsing_notification.py

```diff
@@ -0,0 +1,150 @@
+
+        # Give the browser a little time, because SafeBrowsing.jsm takes a
+        # while between start up and adding the example urls to the db.
+        time.sleep(2)
+
+    def tearDown(self):
+        self.perms.remove('www.itisatrap.org', 'safe-browsing')
+        self.prefs.restore_all_prefs
+        try:
+            self.windows.close_all([self.browser])
+        finally:
+            FirefoxTestCase.tearDown(self)
+
+    def test_notification_bar(self):
+        with self.marionette.using_context("content"):
+            for item in self.urls:
+                label, report_page, unsafe_page = item
+                self.marionette.navigate(unsafe_page)
+                self.check_ignore_warning_button(unsafe_page)
+                self.check_no_phishing_button(label, report_page)
+
+                # Go back to the notification bar
+                self.marionette.navigate(unsafe_page)
+                self.check_ignore_warning_button(unsafe_page)
+
+                # Test the get me out of here button
+                self.check_get_me_out_of_here_button()
+
+                # Go back to the notification bar
+                self.marionette.navigate(unsafe_page)
+                self.check_ignore_warning_button(unsafe_page)
+
+                self.marionette.set_context('chrome')
+                self.check_x_button()
+                self.marionette.set_context('content')
+
+    def check_ignore_warning_button(self, unsafe_page):
+        ignore_warning_button = self.marionette.find_element(By.ID, "ignoreWarningButton")
+
+        # This isn't clickable by the time we get here and needs a
+        # pretty significant delay.
+        time.sleep(1)
+
+        ignore_warning_button.click()
+
+        def find_main_feature_el(mn):
+            try:
+                mn.find_element("id", "main-feature")
+                return True
+            except:
+                return False
+
+        self.wait_for_condition(find_main_feature_el)
+        self.assertRaises(NoSuchElementException, self.marionette.find_element,
+                          'id', 'ignoreWarningButton')
+        self.assertEquals(self.marionette.get_url(), unsafe_page)
+        self.perms.remove('www.itisatrap.org', 'safe-browsing')
+
+    def find_button(self, element, label):
+        buttons = element.find_elements('tag name', 'button')
+        found_button = None
+        for button in buttons:
+            if button.get_attribute('label') == label:
+                found_button = button
+        self.assertTrue(found_button is not None, 'Could not find button')
+        return found_button
+
+    def check_no_phishing_button(self, label, report_page):
+        button_label = self.browser.get_property(label)
+        with self.marionette.using_context('chrome'):
+            tab_count = len(self.tabbar.tabs)
+            blocked_badware_page = (self.marionette.find_element(By.ID, 'content')
```

Comment:
Please see the `tabs_tabPanel` code in the tabs.js module (line 673 ff) for the mozmill-tests. The panel is hard to get, but if you have the tab as reference you can get the panel id which itself can be used to retrieve the panel for this specific tab.